src/lib.rs

```diff
@@ -414,11 +427,11 @@ impl DistinguishedName {
 		removed
 	}
 	/// Inserts or updates an attribute that consists of type and name
-	pub fn push(&mut self, ty :DnType, s :impl Into<String>) {
+	pub fn push(&mut self, ty :DnType, s :DnValue) {
```

Comment:
I'd prefer to keep the old API for the push side, as 99% of use cases would probably be UTF-8 strings. This doesn't just reduce the diff of this PR but also makes updating easier for the users. Usage of the API would also remain as easy as it used to be. Can you rename this function to `push_ext` or something and say something along the lines that you should use `push` unless there is a specific need to use something other than `Utf8Strings`?

The `get` side is fine, as I think it's less used.

Reply:
Would it be better to have a single method that takes an `impl Into<DnValue>` and provide the following implementation?

```rust
impl<T> From<T> for DnValue
where
	T :Into<String>
{
	fn from(t :T) -> Self {
		DnValue::Utf8String(t.into())
	}
}
```

Reply:
> Would it be better to have a single method that takes an impl Into<DnValue> and provide the following implementation?

That's a great idea! I'm a bit worried though that the impl is not immediately seen. A code example in the rustdoc that uses a str literal would solve that I think.